### ISSUE TITLE
Incorrect parsing of query strings

### ISSUE THREAD

**Issue by papercreatures** [radmen/miuri.js#4]
it is perfectly valid to have ?a=1&a=2 unfortunately as it uses a dictionary internally, these get squashed. i'll try and patch next week.

**Comment by radmen** [comment]
So what value should be under `a` key? In PHP `$_GET['a']` will return `2` (Miuri also returns `2`).
I believe it's valid behavior.

**Comment by papercreatures** [comment]
PHP is broken then, but we all knew that.

**Comment by radmen** [comment]
Ok, so what is the correct value then? Can You give something like RFC to show the right way?

**Comment by papercreatures** [comment]
The correct value would be for miuri to not delete duplicate params when parsing. i.e
new Miuri("http://a.com?a=2&a=3') should not become http://a.com?a=2 when toString is called. 

I'm not sure how you would make a backward compatible change to the query function in order to make it possible to set multiples though. jQuery uses a name, value dictionary.

**Comment by radmen** [comment]
> I'm not sure how you would make a backward compatible change to the query function in order to make it possible to set multiples though.

Are You talking about arrays passed in GET (something like `a[]=2&a[]=3`)?

**Comment by papercreatures** [comment]
Yes, but that's just a hacky workaround PHP does. its perfectly valid to do a=1&a=2 sans square brackets

**Comment by radmen** [comment]
I searched a little at Google. You're right about passing arrays without brackets (didn't know that..).
Yet it's not a common thing. jQuery e.g uses brackets (I've used [$.param](http://api.jquery.com/jQuery.param/) to check it).

Parsing query string to find arrays without brackets won't be much a problem. 
Building URI with arrays in query is a little bit tricky, because programmer will have to choose which option (brackets or no brackets) to use.
